Zend/zend_vm_def.h

```diff
@@ -8935,7 +8935,29 @@ ZEND_VM_COLD_CONST_HANDLER(197, ZEND_MATCH_ERROR, CONST|TMPVARCV, UNUSED)
 
 	SAVE_OPLINE();
 	op = GET_OP1_ZVAL_PTR_UNDEF(BP_VAR_R);
-	zend_throw_exception_ex(zend_ce_unhandled_match_error, 0, "Unhandled match value of type %s", zend_zval_type_name(op));
+
+	// For simpler types where there is a convenient stringified version, include that
+	// in the error message. Otherwise just show its type.
+	switch (Z_TYPE_P(op)) {
+		case IS_FALSE:
+			zend_throw_exception_ex(zend_ce_unhandled_match_error, 0, "Unhandled match value: false");
+			break;
+		case IS_TRUE:
+			zend_throw_exception_ex(zend_ce_unhandled_match_error, 0, "Unhandled match value: true");
+			break;
+		case IS_LONG:
+		case IS_DOUBLE:
+		case IS_STRING: {
+			zend_string* stringified = zval_get_string(op);
+			zend_throw_exception_ex(zend_ce_unhandled_match_error, 0, "Unhandled match value: %s", ZSTR_VAL(stringified));
```

Comment:
I don't think this is enough. Strings should be truncated (they could be extremely long). They could also contain newlines that should be handled accordingly.